src/user-statuses/UserStatusScreen.js

```diff
@@ -36,74 +58,136 @@ type Props = $ReadOnly<{|
+  const [textInputValue, setTextInputValue] = useState<string>(
+    inputValueFromStatusText(userStatusText),
+  );
+  const [emojiInputValue, setEmojiInputValue] = useState<EmojiInputValue>(
+    inputValueFromStatusEmoji(userStatusEmoji),
+  );
   const _ = useContext(TranslationContext);
 
   const sendToServer = useCallback(
     partialUserStatus => {
-      api.updateUserStatus(auth, partialUserStatus);
+      const copy = { ...partialUserStatus };
+      // TODO: Put conditional in `api.updateUserStatus` itself; see
+      //   https://github.com/zulip/zulip-mobile/issues/4659#issuecomment-914996061
+      // TODO(server-5.0): Remove conditional (emoji supported as of FL 86:
+      //   https://zulip.com/api/changelog )
+      if (zulipFeatureLevel < 86) {
+        delete copy.status_emoji;
+      }
+      api.updateUserStatus(auth, copy);
       navigation.goBack();
     },
-    [navigation, auth],
+    [zulipFeatureLevel, navigation, auth],
   );
 
-  const handlePressUpdate = useCallback(() => {
-    sendToServer({ status_text: statusText });
-  }, [statusText, sendToServer]);
+  const handlePressSave = useCallback(() => {
+    sendToServer({
+      status_text: statusTextFromInputValue(textInputValue),
+      status_emoji: statusEmojiFromInputValue(emojiInputValue),
+    });
+  }, [textInputValue, emojiInputValue, sendToServer]);
 
   const handlePressClear = useCallback(() => {
-    setStatusText(null);
-    sendToServer({ status_text: null });
-  }, [sendToServer]);
+    setTextInputValue(inputValueFromStatusText(null));
+    setEmojiInputValue(inputValueFromStatusEmoji(null));
+  }, []);
 
   return (
     <Screen title="User status">
-      <Input
-        autoFocus
-        maxLength={60}
-        style={styles.statusTextInput}
-        placeholder="What’s your status?"
-        value={statusText || ''}
-        onChangeText={setStatusText}
-      />
-      <FlatList
-        data={statusSuggestions}
-        keyboardShouldPersistTaps="always"
-        keyExtractor={item => item}
-        renderItem={({ item, index }) => (
-          <SelectableOptionRow
-            key={item}
-            itemKey={item}
-            title={item}
-            selected={item === statusText}
-            onRequestSelectionChange={itemKey => {
-              setStatusText(_(itemKey));
-            }}
+      <View style={styles.inputRow}>
+        {zulipFeatureLevel >= 86 && (
+          <EmojiInput
+            navigation={navigation}
+            value={emojiInputValue}
+            onChangeValue={setEmojiInputValue}
+            rightMargin
           />
         )}
-      />
-      <View style={styles.buttonsWrapper}>
-        <ZulipButton
-          style={styles.button}
-          secondary
-          text="Clear"
-          onPress={handlePressClear}
-          Icon={IconCancel}
-        />
-        <ZulipButton
-          style={styles.button}
-          text="Update"
-          onPress={handlePressUpdate}
-          Icon={IconDone}
```

Comment:
I think this is much more common wording. And the checkmark icon is
    potentially confusing because it looks just like the checkmark on
    one of the status suggestions (SelectableOptionRow), if you've
    selected one.

Yeah. Plus, it doesn't feel normal to me to have an icon at all for a simple "OK" button like this.